packages/application-shell/src/test-utils.js

```diff
@@ -45,30 +81,53 @@ export const render = (
     // flopflip
     adpater = memoryAdapter,
     flags = {},
+    // application-state
+    dataLocale = 'en',
+    environment,
+    user,
+    project,
     // forwarding to react-testing-library
     ...renderOptions
   } = {}
-) => ({
-  ...rtlRender(
-    <IntlProvider locale={locale}>
-      <ApolloMockProvider mocks={mocks} addTypename={addTypename}>
-        <ConfigureFlopFlip adapter={adpater} defaultFlags={flags}>
-          <Router history={history}>{ui}</Router>
-        </ConfigureFlopFlip>
-      </ApolloMockProvider>
-    </IntlProvider>,
-    renderOptions
-  ),
-  // adding `history` to the returned utilities to allow us
-  // to reference it in our tests (just try to avoid using
-  // this to test implementation details).
-  history,
-});
+) => {
+  const mergedUser = mergeOptional(defaultUser, user);
+  const mergedProject = mergeOptional(defaultProject, project);
+  const mergedEnvironment = mergeOptional(defaultEnvironment, environment);
+  return {
+    ...rtlRender(
```

Comment:
Never seen this before 🌠.

Reply:
This is just `render` from `react-testing-library`, which I've renamed in the import. The `rtl` is short for `react-testing-library`. I found this pattern on the `react-testling-library` README and it's awesome.